lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -200,13 +200,13 @@ internal void Reset()
 
         /// <summary>
         /// This method is called when receiving ICompletedTask event during task running or system shutting down.
-        /// Removes the task from running tasks
+        /// Removes the task from running tasks if it was running
         /// Changes the task state from RunningTask to CompletedTask
         /// </summary>
         /// <param name="completedTask"></param>
         internal void RecordCompletedTask(ICompletedTask completedTask)
         {
-            RemoveRunningTask(completedTask.Id);
+            _runningTasks.Remove(completedTask.Id);
```

Comment:
I actually already removed RemoveRunningTask() method. 
At the place where contains is required, I do check it to ensure no mistake. 
At the places where contains is not required as the call may come from different states, I just call  _runningTasks.Remove(taskId); This method returns false it the element is not in the collection but no error.